packages/lwc-language-server/src/javascript/__tests__/compiler.test.ts

```diff
@@ -103,74 +34,35 @@ export default class Foo extends LightningElement {
 }
 `;
 
-it('transform throws exceptions on syntax errors', async () => {
-    try {
-        const options: CompilerOptions = {
-            name: 'foo',
-            namespace: 'x',
-            files: {},
-        };
-        await transform(codeSyntaxError, 'foo.js', options);
-        fail('expects exception');
-    } catch (err) {
-        // verify err has the info we need
-        const message = extractMessageFromBabelError(err.message);
-        expect(message).toMatch('Unexpected token (4:17)');
-        expect(err.location).toEqual({ line: 4, column: 17 });
-    }
+it('can get metadata from a simple component', async () => {
+    await compileSource(codeOk, 'foo.js');
 });
 
-it('transform also throws exceptions for other errors', async () => {
-    const options: CompilerOptions = {
-        name: 'foo',
-        namespace: 'x',
-        files: {},
-    };
-    try {
-        await transform(codeError, 'foo.js', options);
-    } catch (err) {
-        // verify err has the info we need
-        const message = extractMessageFromBabelError(err.message);
-        expect(message).toMatch('Boolean public property must default to false.');
-        const location = extractLocationFromBabelError(err.message);
-        expect(location).toEqual({ line: 5, column: 4 });
-    }
+it('displays an error for a component with syntax error', async () => {
+    const result = await compileSource(codeSyntaxError, 'foo.js');
+    expect(result.metadata).toBeUndefined();
+    expect(result.diagnostics).toHaveLength(1);
+    const [diagnostic] = result.diagnostics;
+    expect(diagnostic.message).toMatch('Unexpected token (4:17)');
 });
```

Comment:
This is a new test that checks for diagnostics on a broken component, and corresponds most closely with the removed test labeled `it('transform throws exceptions on syntax errors')`.